component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/MutualTLSClientAuthenticator.java

```diff
@@ -271,21 +273,38 @@ private Optional<X509Certificate> getCertificateFromHeader(HttpServletRequest re
      * @return X509Certificate X.509 certificate after decoding the certificate content.
      * @throws CertificateException Certificate Exception.
      */
-    private X509Certificate parseCertificate(String content) throws CertificateException {
+    private X509Certificate parseCertificate(String content) throws CertificateException, UnsupportedEncodingException {
 
-        // Trim extra spaces.
-        String decodedContent = StringUtils.trim(content);
-        // Remove Certificate Headers.
-        String certBody = decodedContent.replaceAll(CommonConstants.BEGIN_CERT, StringUtils.EMPTY)
-                .replaceAll(CommonConstants.END_CERT, StringUtils.EMPTY);
-        // Removing all whitespaces and new lines.
-        certBody = certBody.replaceAll("\\s", StringUtils.EMPTY).replace("\\n", StringUtils.EMPTY);
-        byte[] decoded = Base64.getDecoder().decode(certBody);
+        byte[] decoded;
+        String sanitizedCertificate = sanitizeCertificate(content);
+        // First we try to Base64 decode, if it is not decodable, we try to url decode first and then Base64 decode.
+        try {
+            decoded = Base64.getDecoder().decode(sanitizedCertificate);
+        } catch (IllegalArgumentException e) {
+            String urlDecodedContent = URLDecoder.decode(content, StandardCharsets.UTF_8.name());
```

Comment:
Let's add a debug log and log the content. We'll be able to use it for troubleshooting if the fix doesn't work

Reply:
Added with https://github.com/wso2-extensions/identity-oauth-addons/pull/132/commits/c88da558a28ea1a806741447b328214111e6dde2

Reply:
Is it safe to debug log the certificate here?